### DISCUSSION TITLE
comp_barplot reduce/remove space between axes and bars

### DISCUSSION THREAD

**Discussion by greboul** [david-barnett/microViz#167] [Q&A]
Hi,

I have been trying to reduce the space between axes and plot and I just can't figure it out.

It seems that there is some space created by comp_barplot like shown in example "p1" on the function wiki page.
![image](https://github.com/david-barnett/microViz/assets/13840904/ef60a2a9-e787-46b8-a485-7995b3a694f3)

Is there a way to reduce such spacing between bars and axes?

Thanks,
Guillaume

**Comment by david-barnett** [comment]
add expand = FALSE to the coord_flip() or to coord_cartesian() if not flipped

``` r
library(microViz)
#> microViz version 0.12.3 - Copyright (C) 2021-2024 David Barnett
#> ! Website: https://david-barnett.github.io/microViz
#> ✔ Useful?  For citation details, run: `citation("microViz")`
#> ✖ Silence? `suppressPackageStartupMessages(library(microViz))`
```

``` r
library(ggplot2)
data(dietswap, package = "microbiome")

dietswap %>%
  ps_filter(timepoint == 1) %>%
  comp_barplot(tax_level = "Family", n_taxa = 8) + 
  coord_flip(expand = FALSE)
```

![](https://i.imgur.com/xd0yAZU.png)

<sup>Created on 2024-06-13 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
